### DISCUSSION TITLE
Renaming features for the purpose of integration

### DISCUSSION THREAD

**Discussion by nick-youngblut** [satijalab/seurat#8999] [Q&A]
The features in my scRNA-seq dataset are uppercase (e.g., `ZXDB`), while the features in my reference scRNA-seq dataset are capitalized (e.g., `Zxdb`). I thought that it would be trivial to change the reference features to all uppercase, but renaming features appears to be non-trivial for a Seurat object. 

There is a [RenameCells](https://satijalab.github.io/seurat-object/reference/RenameCells.html) function and a [RenameAssays](https://satijalab.github.io/seurat-object/reference/RenameAssays.html) function, but I don't see any `RenameFeatures` function.

Can someone please point me to docs on the "correct" way of renaming features in a Seurat object?

**Comment by wudustan** [comment]
Firstly, it would be good to double check you're working with what you'd expect. The difference in case between gene names would suggest you're working with human (ZXDB) and mouse (Zxdb) genes.

The advice the developers have given previously is to rename your counts matrix before you run `CreateSeuratObject()` or if you're running any of the import functions then you can make sure that the feature names you're importing are the ones you want. 

You can hackily do it by `rownames(x) <- vector.of.rownames` and make sure that the individual layers have been renamed as you need.

**Comment by nick-youngblut** [comment]
> Firstly, it would be good to double check you're working with what you'd expect

Thanks for pointing that out. I was using an old version of the target dataset that used the wrong reference. Still, I would prefer to use accessions as feature names and just switch to gene names (or other naming) when actually needed (e.g., integration). 

> You can hackily do it by rownames(x) <- vector.of.rownames and make sure that the individual layers have been renamed as you need.

Are there currently any docs on renaming features (e.g., renaming via `rownames()` and also dealing with all layers)? It would be great to have a function that dealt with the aspect of renaming all layers.

Thanks for your help!

**Reply by wudustan** [reply]
Not aware of any specific documentation for renaming features. In general, each assay has slots/layers: `counts`, `data` and `scale.data`. When merging multiple datasets, the layers are now stored separately. But, as before, I'd recommend renaming all features before you merge/integrate any data
